source/Engine.cpp

```diff
@@ -504,12 +535,25 @@ void Engine::Step(bool isActive)
 	}
 	else if(flagship)
 	{
-		center = flagship->Center();
-		centerVelocity = flagship->Velocity();
-		Preferences::ExtendedJumpEffects jumpEffectState = Preferences::GetExtendedJumpEffects();
-		if(flagship->IsHyperspacing() && jumpEffectState != Preferences::ExtendedJumpEffects::OFF)
-			centerVelocity *= 1. + pow(flagship->GetHyperspacePercentage() /
-				(jumpEffectState == Preferences::ExtendedJumpEffects::MEDIUM ? 40. : 20.), 2);
+		const auto newCamera = NewCenter(center, centerVelocity,
+			flagship->Position(), flagship->Velocity());
+
+		if(flagship->IsHyperspacing())
+		{
+			center = newCamera.first + (flagship->Position() - newCamera.first) * flagship->GetHyperspacePercentage() / 100.;
```

Comment:
what is going on here?

Reply:
also why are you not using flagship->Center?

Reply:
> what is going on here?

This prevents the camera from behaving strangely during hyperspacing by lerping the camera position to the flagship by how far we are in the hyperspace procedures.

Reply:
> also why are you not using flagship->Center?

Huh, that was added recently, I got used to Body::Position too much

Reply:
> > what is going on here?
> 
> This prevents the camera from behaving strangely during hyperspacing by lerping the camera position to the flagship by how far we are in the hyperspace procedures.

how about you make lerp a function in the namespace?

Reply:
> > > what is going on here?
> > 
> > 
> > This prevents the camera from behaving strangely during hyperspacing by lerping the camera position to the flagship by how far we are in the hyperspace procedures.
> 
> how about you make lerp a function in the namespace?

If we're going to go that rout, I'll make it a method on Point instead.